app.py

```diff
@@ -12,45 +15,210 @@
     initial_sidebar_state="expanded",
 )
 
-st.title("A sneak peek at the stats in Open Targets")
+st.title("A sneak peek at the stats in the Open Targets Platform.")
+st.markdown('##')
 st.markdown("This application is a dashboard to display the metrics of the different data releases in Open Targets.")
-st.sidebar.title("What do you want to check?")
+
+st.sidebar.header("What do you want to do?")
+page = st.sidebar.radio(
+    "Choose an option",
+    ("Explore metrics", "Compare metrics", "Statistics"),
+    index=1,
+    help="Explore metrics allows you to visualise and filter the metrics of the selected datasets. Compare metrics allows you to compare the main metrics between two releases."
+)
 
 # Load data
 files = glob.glob("data/*.csv")
 dfs = [pd.read_csv(f) for f in files]
 data = pd.concat(dfs, ignore_index=True)
 
-# Post or pre pipeline run metrics
-select = st.sidebar.checkbox("Pre pipeline metrics")
-if select:
-    st.subheader('Metrics of the data ran before the pipeline run.')
-    data = data[data["runId"].str.contains("pre")]
+if page == "Explore metrics":
+    # Select a dataset to explore
+    runs = list(data.runId.unique())
+    runs.insert(0, "All")
+    select_run = st.sidebar.selectbox(
+        "Select a pipeline run:",
+        runs
+    )
 
-pivot = pd.pivot_table(data, values="count", index=["variable", "field"], columns="datasourceId")
-st.write(pivot)
+    # Apply select_run mask
+    if select_run != "All":
+        mask_run = data["runId"] == select_run
+        data = data[mask_run]
+    
+    # Refine the query
+    st.markdown("## Filter the data")
+    select_variables = st.multiselect(
+        "Filter by your variables of interest to refine your analysis. You can select as many as you want:",
+        data.variable.unique()
+    )
+    if select_variables:
+        mask_variable = data["variable"].isin(select_variables)
+        data = data[mask_variable]
+        if any("Field" in variable for variable in select_variables):
+            select_fields = st.multiselect(
+                "You can also filter by your field of interest:",
+                data.field.unique()
+            )
+            mask_field = data["field"].isin(select_fields)
+            data = data[mask_field] if select_fields else data.copy()
 
-st.sidebar.title("Choose a pipeline run")
-release = st.sidebar.selectbox(
-    'Data release',
-    data.runId.unique()
-)
+    # Generate pivot table
+    select_pivot = st.checkbox("See pivot table.")
+    if select_pivot:
+        try:
+            # Custom pivot table to avoid loss of data when field == NaN
+            output = data.set_index(["variable", "field", "datasourceId"]).unstack("datasourceId", fill_value=0).drop("runId", axis=1)
```

Comment:
For wrapping the line length to <120 characters, & also for readability:

```suggestion
            output = (
                data
                .set_index(["variable", "field", "datasourceId"])
                .unstack("datasourceId", fill_value=0)
                .drop("runId", axis=1)
            )
```

Reply:
Fixed in `025ba22`